controllers/datadogagent/override/global.go

```
@@ -6,26 +6,199 @@
+	if config.ClusterName != nil {
+		manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+			Name:  apicommon.DDClusterName,
+			Value: *config.ClusterName,
+		})
+	}
+
+	// Site is the Datadog intake site Agent data are sent to.
+	manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+		Name:  apicommon.DDSite,
+		Value: *config.Site,
+	})
 
-	if config != nil && config.Kubelet != nil && config.Kubelet.TLSVerify != nil {
-		manager.EnvVar().AddEnvVarToContainer(apicommonv1.CoreAgentContainerName, &corev1.EnvVar{
-			Name:  apicommon.DDKubeletTLSVerify,
-			Value: apiutils.BoolToString(config.Kubelet.TLSVerify),
+	// Endpoint is the Datadog intake URL the Agent data are sent to.
+	if config.Endpoint != nil && config.Endpoint.URL != nil {
+		manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+			Name:  apicommon.DDddURL,
+			Value: *config.Endpoint.URL,
 		})
 	}
 
-	// set image registry
+	// Registry is the image registry to use for all Agent images.
+	for _, c := range manager.PodTemplateSpec().Spec.InitContainers {
+		c.Image = *config.Registry
+	}
+	for _, c := range manager.PodTemplateSpec().Spec.Containers {
+		c.Image = *config.Registry
+	}
+
+	// LogLevel sets logging verbosity. This can be overridden by container.
+	manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+		Name:  apicommon.DDLogLevel,
+		Value: *config.LogLevel,
+	})
+
+	// NetworkPolicy contains the network configuration.
+	if config.NetworkPolicy != nil {
+		if apiutils.BoolValue(config.NetworkPolicy.Create) {
+			switch config.NetworkPolicy.Flavor {
+			case v2alpha1.NetworkPolicyFlavorKubernetes:
+				_ = resourcesManager.NetworkPolicyManager().BuildKubernetesNetworkPolicy(dda, componentName)
+			case v2alpha1.NetworkPolicyFlavorCilium:
+				// TODO
+			}
+		}
+	}
+
+	if componentName == v2alpha1.NodeAgentComponentName {
+		// Tags contains a list of tags to attach to every metric, event and service check collected.
+		if config.Tags != nil {
+			tags, _ := json.Marshal(config.Tags)
```

I don't think we should ignore errors in this `json.Marshal` call and the others in this function. These config values can come from the user and nothing guarantees that they're going to be valid json.